packages/bits/demo/src/components/demo/wizard-v2/wizard-restore-state/wizard-restore-state.example.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, Inject, OnInit, QueryList, TemplateRef } from "@angular/core";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { DialogService, IWizardState, NuiDialogRef, WizardStepV2Component } from "@nova-ui/bits";
+
+@Component({
+    selector: "nui-wizard-restore-state-example",
+    templateUrl: "./wizard-restore-state.example.component.html",
+})
+export class WizardRestoreStateExampleComponent implements OnInit {
+    public form: FormGroup;
+    public activeDialog: NuiDialogRef;
+    public stateToRestore: IWizardState;
+
+    constructor(private formBuilder: FormBuilder,
+                @Inject(DialogService) private dialogService: DialogService) {}
+
+    ngOnInit(): void {
+        this.form = new FormGroup({
+            "personDetails": this.formBuilder.group({
+                "name": [
+                    "",
+                    [Validators.required, Validators.minLength(3)],
+                ],
+                "privacy": ["", [Validators.required]],
+            }),
+            "organization": this.formBuilder.group({
+                "title": ["", [Validators.required]],
+                "date": ["", [Validators.required]],
+            }),
+            "contactDetails": this.formBuilder.group({
+                "email": ["", [Validators.required, Validators.email]],
+                "options": [""],
+            }),
+        });
+    }
+
+    public openDialog(content: TemplateRef<string>): void {
+        this.activeDialog = this.dialogService.open(content, {size: "lg"});
+    }
+
+    public closeDialog(): void {
+        this.activeDialog.close();
+    }
+
+    public onWizardDestroy(steps: QueryList<WizardStepV2Component>): void {
```

Comment:
Why is this callback called `onWizardDestroy`?

Reply:
curranty, we emitting steps list (which we can restore later)  onDestroy hook because it most common case for users.